components/FlexibleHeader/src/MDCFlexibleHeaderView.m

```diff
@@ -635,6 +635,13 @@ - (CGFloat)fhv_enforceInsetsForScrollView:(UIScrollView *)scrollView {
                       && [self trackingScrollViewIsWebKit])) {
     return 0;
   }
+  if (@available(iOS 11.0, *)) {
+    // Don't adjust the contentInset if scrollView's behavior doesn't want it.
+    // Compatible to iOS 11 and above
+    if (scrollView.contentInsetAdjustmentBehavior == UIScrollViewContentInsetAdjustmentNever) {
+      return 0;
+    }
+  }
```

Comment:
This looks like a risky behavioral change that we'll want to gate behind a runtime flag.

Please add a public BOOL property that is NO by default and, when enabled, enables this new code path. This will led us gradually migrate clients to the new behavior so that we can resolve any dependency on the prior behavior.

Reply:
Sure. What do you think would be a suggested name for this property? I am not sure if there is a convention of the flag names for migration. Thanks!

Reply:
No formal conventions exist yet, though that's probably a good idea for us to explore as a team decision. Filed https://github.com/material-components/material-components-ios/issues/5154 to track.

```
contentInsetAdjustmentBehavior == UIScrollViewContentInsetAdjustmentNever
```

Some possible suggestions:

- disableContentInsetAdjustmentWhenContentInsetAdjustmentBehaviorIsNever
- respectsContentInsetAdjustmentNever

Reply:
Thanks for the suggestion and creating the ticket to track. I will use the first one for now.